### ISSUE TITLE
processPdf misclassifies a correctly-password-decrypted RC4 V2/R3/128 PDF as Scanned (0 chars extracted)

### ISSUE THREAD

**Issue by chintans** [firecrawl/pdf-inspector#524]
## Summary

A real-world RC4-encrypted PDF (`/Filter /Standard /V 2 /R 3 /Length 128`) decrypts *structurally* with the correct password — `processPdf` does not throw `Encrypted`, and the reported page count matches the unencrypted original exactly — but the content streams appear to stay undecoded: `pdf_type` comes back `Scanned` with every page flagged in `pages_needing_ocr`, `markdown` is empty, and `confidence` is reported as high (~0.95) despite the extraction being completely empty.

The same file, manually decrypted (password removed via a desktop PDF reader) and re-saved, classifies as `TextBased` with confidence 1.0 and extracts ~65k characters of clean Markdown. So the document genuinely has a normal text layer — decrypting it programmatically with the correct password is what's failing to reach the content streams.

## Why this looks related to, but distinct from, #138

[#138](https://github.com/firecrawl/pdf-inspector/pull/138) fixed a bug with an almost identical description: *"some encrypted PDFs load structurally via `load_mem` with `is_encrypted()` still `true` — the streams stay encrypted, so the extracted 'text' is garbage and the file gets misclassified as Scanned."* That fix is already in the version I'm running (1.18.0, well after #138 merged on 2026-07-11), so this isn't the same code path — it looks like a different case within the same failure family, plausibly in how `lopdf` (the underlying parser) handles this particular RC4 key length/revision combination once `load_mem_with_options(..., LoadOptions::with_password(pw))` is used.

## Environment

- `@firecrawl/pdf-inspector-wasm` `1.18.0`, reproduced directly under Node 24 (`initSync` + `processPdf`) on Windows.
- My production setup also calls the equivalent Rust API (`process_pdf_mem_with_options` with `PdfOptions.password` set, same crate version) from a small Axum service — that service rejects the same document with the same "no text found" outcome, consistent with this bug, though I've only isolated a minimal standalone repro through the WASM binding below.

## Repro

I can't attach the original file — it's a real customer's health insurance policy document — but the encryption parameters are directly visible in the trailer without decryption:

```
175 0 obj<</O (...)/P -1852/R 3/Filter/Standard/U (...)/V 2/Length 128>>
```

i.e. Standard security handler, `V=2`, `R=3`, 128-bit RC4 key length (non-default `/Length`).

```js
import { initSync, processPdf } from "@firecrawl/pdf-inspector-wasm";
// ... initSync(...)

const bytes = new Uint8Array(fs.readFileSync("encrypted.pdf"));
const result = processPdf(bytes, { password: "<the correct password>" });

console.log(result.pdfType);         // "Scanned"  (expected: "TextBased")
console.log(result.confidence);      // ~0.95
console.log(result.pagesNeedingOcr); // every page in the document
console.log(result.markdown?.length ?? 0); // 0
```

Decrypting the same bytes with the same password via a standard PDF reader, then re-running `processPdf` with no password on the resulting plaintext PDF, gives:

```
pdfType: "TextBased", confidence: 1, pagesNeedingOcr: [], markdown.length: ~65000
```

## Expected behavior

- When a password is accepted (no `Encrypted` error, correct page count), the content streams should decrypt and extract the same as the unencrypted equivalent.
- Failing that, a confident `Scanned` classification with an accepted password and zero extracted text is a bad failure mode for callers routing on `pdf_type`/`confidence` — this is the same "confidently wrong" category as #272 / #443, just triggered by password decryption rather than font/CJK handling.

## Happy to help

I can't share the source file (real personal health data), but I can build and share a synthetic/redacted PDF with the same `/V 2 /R 3 /Length 128` encryption parameters if that would help reproduce and fix this, or run a debug build against the real file locally and report back specifics.

**Comment by chintans** [comment]
Update: found the actual root cause, and it's one layer deeper than I first
thought — it's in `lopdf` (pdf-inspector's own dependency), not in
pdf-inspector's code.

The document I filed this against has *distinct* owner and user passwords
(case variants of the same string). Opening it with the owner password
authenticates successfully in `lopdf` (no `Encrypted` error), but `lopdf`
0.44.0 then derives the file encryption key from the *literal owner
password* instead of the *user password* Algorithm 7 recovers from `/O` —
which is what Algorithm 2 actually requires. Decryption "succeeds" with no
error (dictionaries, names, integers and references are never encrypted, so
structure and page count parse correctly regardless), but every decrypted
string and stream — including the text pdf-inspector extracts — comes out
as garbage, which is exactly what got misclassified here as `Scanned`.

Filed and fixed upstream: https://github.com/J-F-Liu/lopdf/pull/573

Once that merges and a release picks it up, bumping the `lopdf` dependency
here should resolve this without any change needed in pdf-inspector itself.
